### PR TITLE
Bug 250 layout editors centering gravity doesnt work

### LINKED ISSUE #250: Layout editor's centering gravity doesn't work
As a user I'd like to center layout using layout gravity

### PR DESCRIPTION
**Pull request by terolamminpaa**
resolves #250

### CODE REVIEW

**Review by anttileppa — CHANGES_REQUESTED** [review]
Fix sonar issues

File: src/components/preview/components/page-preview-frame-layout.tsx [hunk 1]
```diff
@@ -160,11 +160,21 @@ class PagePreviewFrameLayout extends React.Component<Props, State> {
           break;
           case "layout_gravity":
             property.value.split("|").forEach(gravityValue => {
+              const layoutWidth = parseFloat(String(result.width));
```
Comment (anttileppa) [inline src/components/preview/components/page-preview-frame-layout.tsx:163]: This depends on the order of properties and it is not guaranteed to be always same. Instead you need to resolve width and height from properties, not from result

**Review by anttileppa — CHANGES_REQUESTED** [review]

File: src/components/preview/components/page-preview-frame-layout.tsx (hunk 1 above)
Comment (terolamminpaa) [reply src/components/preview/components/page-preview-frame-layout.tsx]: There is an entirely new implementation which uses flex and doesn't require width and height calculations

File: src/components/preview/components/page-preview-frame-layout.tsx [hunk 2]
```diff
@@ -65,12 +68,12 @@ class PagePreviewFrameLayout extends React.Component<Props, State> {
...
     const result = (this.props.view.children || []).map((child: PageLayoutView, index: number) => {
-      return <PagePreviewComponentEditor key={ `child-${index}` } 
+      return <PagePreviewComponentEditor key={ `child-${index}` }
         view={ child }
         resourceMap={ this.props.resourceMap }
-        displayMetrics={ this.props.displayMetrics } 
-        scale={ this.props.scale }        
-        handleLayoutProperties={ this.onHandleLayoutProperties }/>
+        displayMetrics={ this.props.displayMetrics }
+        scale={ this.props.scale }
```
Comment (jarinykanen) [inline src/components/preview/components/page-preview-frame-layout.tsx:75]: Move this.props variables to const { ... } = this.props;

File: src/components/preview/components/page-preview-frame-layout.tsx [hunk 3]
```diff
@@ -115,7 +125,31 @@ class PagePreviewFrameLayout extends React.Component<Props, State> {
...
+  }
+
+  /**
+   * Resolves component styles
+   *
+   * @returns component styles
+   */
+  private resolveStyles = (): CSSProperties => {
+    const properties = this.props.view.properties;
+    const result: CSSProperties = this.props.handleLayoutProperties(properties, {});
```
Comment (jarinykanen) [inline src/components/preview/components/page-preview-frame-layout.tsx:140]: Move this.props variables to const { ... } = this.props;

**Review by jarinykanen — CHANGES_REQUESTED** [review]
Fix comments

**Pull request merged**